src/utils/yara.js

```diff
@@ -0,0 +1,146 @@
+import yara from 'yara';
+
+export default class YaraManager {
+  constructor() {
+    this.resources = [];
+  }
+
+  /*
+   * This Method scans a list of raw js artifacts using
+   * a set of Yara rules. Returns the Js artifacts that match
+   * otherwise, no malware was detected.
+   *
+   * If an error occurs in processing, values are pushed to the RO
+   * in order to be handled correctly by our api.
+   *
+   * When pushing errors to the RO data buffer:
+   *      error: boolean (whether or not an error has occurred in processing)
+   *      matchedRule: only valid if (isMalicious == true)
+   *      isMalicious: boolean, null if unknown
+   */
+  setupResourceScan(jsArtifacts, ticket) {
+    console.log('Scanning JS...');
+    const ticketId = ticket.getID();
+
+    yara.initialize(error => {
+      if (error) {
+        console.log(error.message);
+
+        this.resources.push({
+          ticketId,
+          filename: 'yara_error.txt',
+          data: Buffer.from(
```

Comment:
Small improvement would be to make a block-level or `YaraManager` level helper function that makes all this for you, just takes parameters if they vary based on where the error happens.

Reply:
Is this what you were thinking?